src/Compilers/Core/Portable/CodeGen/ILBuilderEmit.cs

```diff
@@ -66,6 +66,11 @@ internal void EmitModuleVersionIdStringToken()
             this.GetCurrentWriter().WriteUInt32(Cci.MetadataWriter.ModuleVersionIdStringToken);
         }
 
+        internal void EmitSourceDocumentIndexToken(Cci.DebugSourceDocument document)
+        {
+            this.GetCurrentWriter().WriteUInt32((module?.GetFakeSourceDocumentTokenForIL(document) ?? 0xFFFF) | Cci.MetadataWriter.SourceDocumentIndex);
```

Comment:
I found this value--as a literal--preexisting in the EmitToken logic. I don't know for sure what it means--my guess is that it's supposed to be some sort of recognized invalid value. If that is the case, then the value seems too low--it's easy to imagine needing more than 65536 tokens per assembly.

Does anyone know what this means and why the value is what it is?